### PR TITLE
Update mjml extensions to be written in typescript

### PR DESCRIPTION
**Pull request by hchiufaire**
Addresses https://github.com/Faire/mjml-react/issues/22, converting src/extensions folder to typescript.

### Tests
image

### CODE REVIEW

File: src/extensions/mjml-tracking-pixel.tsx [hunk 1]
```diff
@@ -16,13 +20,7 @@ export class MjmlTrackingPixel extends Component {
     return (
       <MjmlRaw>
         {/* eslint-disable-next-line jsx-a11y/alt-text */}
-        <img
-          src={src}
-          style={trackingPixelStyle}
-          width={1}
-          height={1}
-          border={0}
```
Comment (hchiufaire) [inline src/extensions/mjml-tracking-pixel.tsx:24]: Seems `border` is no longer a valid property; the default is `none` anyway but is set as 0 in `trackingPixelStyle`

https://rocketvalidator.com/html-validation/the-border-attribute-is-obsolete-consider-specifying-img-border-0-in-css-instead

File: src/extensions/mjml-html.tsx [hunk 2]
```diff
@@ -1,6 +1,11 @@
 import React, { Component } from "react";
 
-export class MjmlHtml extends Component {
+interface IMjmlHtml {
+  tag: string;
```
Comment (emmclaughlin) [inline src/extensions/mjml-html.tsx:4]: Should this be optional? since there is an mj-raw fallback

**Review by emmclaughlin — COMMENTED** [review]
Can these be just .ts files instead of .tsx? I don't think there is JSX in any of them

**Comment by hchiufaire** [comment]
It seems only [mjml-comment.ts](https://github.com/Faire/mjml-react/blob/a75f97fd6a19b8ecca740103ba62718a2cb7f366/src/extensions/mjml-comment.ts) and [mjml-html.ts](https://github.com/Faire/mjml-react/blob/a75f97fd6a19b8ecca740103ba62718a2cb7f366/src/extensions/mjml-html.ts) don't return jsx so could only rename those

**Review by emmclaughlin — APPROVED** [review]
Looks good

**Pull request merged**